src/main/java/com/sda/games/wheelOfFortune/methods/WheelGame.java

```diff
@@ -219,16 +222,18 @@ private static int generateRNCategory() {
         CategoryDao categoryDao = new CategoryDao(WheelGame.hibernateFactory);
         int max = categoryDao.getAllCount();
         Random random = new Random();
-        return random.nextInt(max+MIN)-MIN;
+        //return random.nextInt(max-MIN)+MIN;
+        return random.nextInt(max)+MIN; //eliminacja błędu ujemnej kategorii
     }
     private static int generateRNWord(int drawnNumberCategory) {
         WordsDao wordsDao = new WordsDao(WheelGame.hibernateFactory);
-        if(drawnNumberCategory<=0){     //
+/*        if(drawnNumberCategory<=0){     //
             drawnNumberCategory=1;      // don't know y but this number can be -1 or 0 (problem with random?)
-        }                               //
+        }  */                             //
         int max = wordsDao.getAllCountWordsFromCategory(drawnNumberCategory);
         Random random = new Random();
-        return  random.nextInt(max+MIN)-MIN;
+        //return random.nextInt(max+MIN)-MIN;
+        return (random.nextInt(max)+MIN)+((drawnNumberCategory-1)*10); //metoda łoma
```

Comment:
to wyglada dosc dziwnie jak jakies obejscie :D 
jesli potrzebujesz 2 roznych danych (randomowe slowo i  ten licznik od kategorii) 

ale tak ogolnie to robicie to baaaardzo na okolo i strasznie kombinujecie :D 
nie kumam czemu metoda `getAllCountWordsFromCategory` zwraca inta ?  przeciez i tak tam w srodku robicie wielkiego selecta wszystkich obiektow. Nie lepiej zwrocic liste obiektow i wybrac sobie z tej listy randomowo jeden ? (takie podejscie bedzie odporne na sytuacje gdzie w id wkradnie sie "dziura" i zapomnicie jakiegos id dodac) 
a jak juz musicie tak robic to zoptymalizujcie te zapytanie zeby zwracalo wam samego inta a nie wszystkie obiekty (jakis count w zapytaniu czy cos :D )